Source/displayinfo/include/ExtendedDisplayIdentification.h

```diff
@@ -303,23 +734,139 @@ namespace Plugin {
+        // Use of the EDID Preferred Timing bit shall be used to indicate that the timing indeed
         // conforms to this definition.
         uint16_t PreferredWidthInPixels() const {
             return IsValid() ? (((_base[0x3A] & 0xF0 ) << 4) + _base[0x38]) : 0;
         }
-        
+
         uint16_t PreferredHeightInPixels() const {
             return IsValid() ? (((_base[0x3D] & 0xF0 ) << 4) + _base[0x3B]) : 0;
         }
 
+        uint8_t SupportedColorDepths() const {
+            uint8_t colorDepthMap = static_cast<uint8_t>(ColorDepth());
+
+            Iterator segment = Iterator(_segments);
+            while(segment.Next() == true) {
+                if(segment.Type() == CEA::extension_tag) {
+                    CEA cae(segment.Current());
+                    colorDepthMap = cae.SupportedColorDepths();
+                    break;
+                }
+            }
+            return colorDepthMap;
+        }
+
+        colorformattype SupportedColorFormat() const {
+            colorformattype result = colorformattype::UNDEFINED;
+
+            Iterator segment = Iterator(_segments);
+            while(segment.Next() == true) {
+                if(segment.Type() == CEA::extension_tag) {
+                    CEA cae(segment.Current());
+                    result = cae.SupportedColorFormat();
+                    break;
+                }
+            }
+
+            return result;
+        }
+
+        uint8_t SupportedDigitalDisplayTypes() const {
+            uint8_t colorFormatMap = static_cast<uint8_t>(colorformattype::RGB);
+
+            if((_base[0x14] >> 7) & 0x01){
+                switch((_base[0x18] >> 3) & 0x07) {
+                    case 0x00:
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::RGB);
+                        break;
+                    case 0x01:
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::RGB);
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::YCBCR_4_4_4);
+                        break;
+                    case 0x02:
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::RGB);
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::YCBCR_4_2_2);
+                        break;
+                    case 0x03:
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::RGB);
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::YCBCR_4_2_2);
+                        colorFormatMap |= static_cast<uint8_t>(colorformattype::YCBCR_4_4_4);
+                        break;
+                }
+            }
+            return colorFormatMap;
+        }
+
+        uint8_t SupportedColorFormats() const {
+            uint8_t colorFormatMap = 0;
+
+            Iterator segment = Iterator(_segments);
+            while(segment.Next() == true) {
```

Comment:
I see a repetition of 6 times the same code to find an CEA::extension_tag. Might be good to move this to method, before every one starts to deploy this copy-Paste OO :-)

Reply:
Done :)